### ISSUE TITLE
RegistryEvent$NewRegistry is removed between patch versions 40.0.15 and 40.0.18

### ISSUE THREAD

**Issue by DuckyCrayfish** [MinecraftForge/MinecraftForge#8519]
**Minecraft Version:** 1.18.2

**Forge Version:** 40.0.18

**Description of issue:**
Despite versions 40.0.15 and 40.0.18 only differing in patch version, the inner class `NewRegistry` is removed from the `RegistryEvent` class between these versions, breaking compatibility for mods depending on it. 

**Known affected mods:**
- Hourglass
- Simple Planes

**Comment by AterAnimAvis** [comment]
We're inside breaking changes period, this is expected, until a `40.1.0` build is published breaking changes **may** occur.

The change in API in this case is down to https://github.com/MinecraftForge/MinecraftForge/pull/8495, which is adapting the ForgeRegistries due to internal changes from Mojang

**Comment by DuckyCrayfish** [comment]
Ahh, understood. While I don't agree with breaking semantic versioning, I understand why it's done in the case of MinecraftForge.

Thanks!

**Comment by marchermans** [comment]
We simply don't follow SemVer.

**Comment by AterAnimAvis** [comment]
Forge doesn't follow Semantic Versioning, it uses the follow schema
`X.Y.Z`
`X` - Major Version, bumped per Minecraft Version (Including minor Minecraft bumps)
`Y` - Current Recommended Build number - considered as "checkpoint" releases, untill the first RB is published Breaking Changes may occur 
`Z` - Commits since last RB

Forge is developed against Minecraft, a moving target (which also doesn't follow SemVer), we require time to be able to adapt and break to keep in line with the changes upstream by Mojang, Semantic Versioning just doesn't cover our use case.

**Comment by DuckyCrayfish** [comment]
Right, sorry for the misunderstanding. 

I guess what I meant by that is "I am of the opinion that it would be quite useful to have a versioning schema that could isolate breaking changes, but I understand how that would be difficult given the constraints of this project."

Thank you for providing me with info on the current strategy though 👍

**Comment by gigaherz** [comment]
We do have a semantic versioning schema. And it does indicate the breaking changes window: We allow breaking changes when the middle component is 0.

First component tags the MC version, and indicates that all previous assumptions are to be reevaluated, since we rebased on top of a new Minecraft version.
Second component indicates the "stability milestone". x.0 -> unstable/beta. X.1 -> we have decided that it's stable enough to release a recommended build, which means anything older than it is to be considered obsolete/unstable.
Third component indicates development progress: bigger number = newer build.

It IS semantic, it's just not following the rules of semver specifically.
